File: 1557-minimum-number-of-vertices-to-reach-all-nodes/1557-minimum-number-of-vertices-to-reach-all-nodes.cpp
class Solution {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        vector<int> deg(n,0);
        for(auto e:edges){
            deg[e[1]] += 1;
        }
        vector<int> ans;
        for(int i=0;i<n;i++){
            if(deg[i] == 0)
                ans.push_back(i);
        }
        return ans;
    }
};

File: 1557-minimum-number-of-vertices-to-reach-all-nodes/1557-minimum-number-of-vertices-to-reach-all-nodes.java
class Solution {
    public List<Integer> findSmallestSetOfVertices(int n, List<List<Integer>> edges) {
        int degree[] = new int[n];
        for(int i=0;i<edges.size();i++){
            degree[edges.get(i).get(1)] += 1;
        }
        
        List<Integer> result = new ArrayList<>();
        for(int i=0;i<n;i++){
            if(degree[i] == 0){
                result.add(i);
            }
        }
        return result;
    }
}